src/ucp/wireup/select.c

```diff
@@ -874,7 +874,8 @@ static ucs_status_t ucp_wireup_add_am_bw_lanes(ucp_ep_h ep, const ucp_ep_params_
     found         = 0;
     tl_bitmap     = -1;
     remote_md_map = -1;
-    for (; found < ep->worker->context->config.ext.max_eager_lanes;) {
+    /* 1 lane is always AM lane */
```

Comment:
why we need that? am_bw_lane[0] would still use same resource as am_lane, if it's good for bw

Reply:
there are issues in decision of what protocol should be used: in case if am_bw[0] is not same as am lane - then there is no simple way to select appropriate thresholds for single/multiple proto

Reply:
but then wouldn't we use am_lane twice in the round robin?
can we first check if we can use "single" protocol on am_lane , and if not, check "single" on am_bw_lane[0], and if not use "multi" on am_bw_lane

Reply:
it will cause significant code duplication - make all checks twice and make logic to call bcopy single with different parameters (there are lot of locations where such logic is implemented)

Reply:
need to solve code duplication somehow without reusing am_lane more than once per round

Reply:
I will update wireup to prevent double AM lane